### PR TITLE
[WEB-7353] Add new Rewards Service endpoints to SWS JS SDK

### PR DESCRIPTION
**Pull request by gabrieljo**

### CODE REVIEW

File: src/Rewards.js [hunk 1]
```diff
@@ -64,4 +64,93 @@ export default class RewardsService extends Service {
...
+  }
+
+  /**
+   * Returns information about a Referee’s usage of the Referral Code
+   * @param {Object} params       - Input parameter object
+   * @param {String} params.code  - Referral code
+   * @param {Int} params.userId  - Referral code
+   * @returns {Promise}
+   */
+  getUsageOfReferralCode ({ code, userId }) {
```
Comment (Mike-Serato) [inline src/Rewards.js:118]: Can be renamed to getRefereeEligibilityByReferralCode

File: src/Rewards.js [hunk 2]
```diff
@@ -64,4 +64,93 @@ export default class RewardsService extends Service {
...
+   * Returns Logs a Referral Campaign activity
+   * @param {Object} params             - Input parameter object
+   * @param {String} params.code        - Referral code
+   * @param {Int} params.referrerUserId - Referral user id
+   * @param {Int} params.refereeUserId  - Referee user id
+   * @param {String} params.voucherId   - Voucher id
+   * @param {String} params.productId   - Product type id
+   * @return {Promise}
+   */
+  addReferralCampaignActivityLog ({
```
Comment (Mike-Serato) [inline src/Rewards.js:137]: There are changes to this endpoint in terms of request body parameters. Please refer to the latest work on WEB-7292.

File: src/Rewards.js [hunk 3]
```diff
@@ -64,4 +64,93 @@ export default class RewardsService extends Service {
...
+    )
+  }
+
+  /**
+   * Returns information about a Referrer’s participation in a Referral Campaign
+   * @param {Object} params - Input parameter object
+   * @param {Int} params.id - Campaign id
+   * @returns {Promise}
+   */
+  getReferrerParticipation ({ id }) {
```
Comment (Mike-Serato) [inline src/Rewards.js:102]: Rename to getReferrerCampaignDetailsById?

File: package-lock.json [hunk 4]
```diff
@@ -1,5224 +1,8 @@
 {
   "name": "@serato/sws-sdk",
   "version": "6.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
   "requires": true,
```
Comment (Mike-Serato) [inline package-lock.json:5]: Any reasons why there is such a huge changes in the lock file?

Comment (gabrieljo) [reply package-lock.json:5]: Have we changed the NPM version? lockfileVersion: 2 is NPM 7. I am still in NPM 6.

File: src/Rewards.js [hunk 5]
```diff
@@ -64,4 +64,97 @@ export default class RewardsService extends Service {
...
+    return this.fetch(
+      this.bearerTokenAuthHeader(),
+      `/referralcode/${code}/referee/${userId}`,
+      null,
+      'GET'
+    )
+  }
+
+  /**
+   * Returns Logs a Referral Campaign activity
```
Comment (Mike-Serato) [inline src/Rewards.js:128]: This comment can be updated to "Adds a log for a referral campaign based on referrer and referee activity"

File: test/integration/RewardsTest.js [hunk 6]
```diff
@@ -39,5 +39,73 @@ describe('Rewards Tests', function () {
...
+          err => {
+            expect(err.httpStatus).not.to.equal(404)
+          }
+        )
+      }
+    )
+
+    it(`confirms URI used in 'getUsageOfReferralCode()' method with Campaign Id, by returning a non-404 HTTP response`,
+      function () {
+        return swsClient.rewards.getUsageOfReferralCode({code: 'testcode', userId: 1}).then(
```
Comment (Mike-Serato) [inline test/integration/RewardsTest.js:89]: There is no such method in the library? Might wanna update these and typescript definitions.

Comment (gabrieljo) [reply test/integration/RewardsTest.js:89]: I will rewrite this test file. it still needs that all endpoint should be on prod.

Comment (Mike-Serato) [reply test/integration/RewardsTest.js:89]: They are all in production now

**Review by Mike-Serato — APPROVED** [review]
A minor thing I noticed otherwise looks good.

**Pull request merged**